### PR TITLE
Move capsules to the data stream

### PR DESCRIPTION
**Pull request by DavidSchinazi**

### CODE REVIEW

File: draft-ietf-masque-h3-datagram.md [hunk 1]
```diff
@@ -195,41 +195,93 @@ silently or buffer it temporarily while awaiting the creation of the
...
+associated streams without any modification of the data stream.
+
+Participant mode:
+
+: In this mode, the intermediary terminates the data stream and parses all
+Capsule Type and Capsule Length fields it receives.
+
+Each Capsule Type determines whether it is opaque or transparent to
+intermediaries in participant mode: opaque capsules are forwarded unmodified
+while transparent ones can be parsed, added, or removed by intermediaries.
```
Comment (LPardue) [inline draft-ietf-masque-h3-datagram.md:284]: assume "modification" of a capsule is really remove and add?

File: draft-ietf-masque-h3-datagram.md [hunk 2]
```diff
@@ -195,41 +195,93 @@ silently or buffer it temporarily while awaiting the creation of the
...
-Capsule Type determines whether it is opaque or transparent to intermediaries:
-opaque capsules are forwarded unmodified while transparent ones can be parsed,
-added, or removed by intermediaries.
 
-This specification of CAPSULE currently uses HTTP/3 frame type 0xffcab5. If this
-document is approved, a lower number will be requested from IANA.
+## Capsule Protocol {#capsule-protocol}
+
+This specification defines the "data stream" of an HTTP request as the
+bidirectional stream of bytes that follow the headers in both directions. In
```
[the remark this answers is not in the document]
Comment (DavidSchinazi) [reply draft-ietf-masque-h3-datagram.md:209]: Done

File: draft-ietf-masque-h3-datagram.md [hunk 3]
```diff
@@ -195,41 +195,93 @@ silently or buffer it temporarily while awaiting the creation of the
...
+If the definition of an HTTP Method or HTTP Upgrade Token states that it uses
+the capsule protocol, its implementations MUST follow the following
+requirements:
+
+* A server MUST NOT send any Transfer-Encoding or Content-Length header fields
+  in a 2xx (Successful) response. If a client receives a Content-Length or
+  Transfer-Encoding header fields in a successful response, it MUST treat that
+  response as malformed.
+
+* A request message does not have content.
```
[the remark this answers is not in the document]
Comment (DavidSchinazi) [reply draft-ietf-masque-h3-datagram.md:267]: I added a note about response, we can add more headers in a followup?

File: draft-ietf-masque-h3-datagram.md (hunk 1 above)
Comment (DavidSchinazi) [reply draft-ietf-masque-h3-datagram.md:289]: Clarified that modification of the data of transparent capsules is OK, after all it's isomorphic to remove + add

File: draft-ietf-masque-h3-datagram.md [hunk 4]
```diff
@@ -195,41 +195,93 @@ silently or buffer it temporarily while awaiting the creation of the
...
-added, or removed by intermediaries.
 
-This specification of CAPSULE currently uses HTTP/3 frame type 0xffcab5. If this
-document is approved, a lower number will be requested from IANA.
+## Capsule Protocol {#capsule-protocol}
+
+This specification defines the "data stream" of an HTTP request as the
+bidirectional stream of bytes that follow the headers in both directions. In
+HTTP/1.x, the data stream consists of all bytes on the connection that follow
+the blank line that concludes the header section in both directions. In HTTP/2
```
[the remark this answers is not in the document]
Comment (DavidSchinazi) [reply draft-ietf-masque-h3-datagram.md]: Fixed

**Comment by DavidSchinazi** [comment]
Had offline conversation with Lucas, conclusion was that this is good enough to merge to allow interop - we'll have plenty of time to figure out details in subsequent issues and PRs.

**Pull request merged**